api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/MetricsServiceLifecycle.java

```diff
@@ -0,0 +1,241 @@
+        // Create the shared instance now and initialize it with the C* session when ready
+        metricsService = new MetricsServiceImpl();
+        ThreadFactory threadFactory = r -> {
+            Thread thread = Executors.defaultThreadFactory().newThread(r);
+            thread.setName(MetricsService.class.getSimpleName().toLowerCase(Locale.ROOT) + "-lifecycle-thread");
+            return thread;
+        };
+        // All lifecycle operations will be executed on a single thread to avoid synchronization issues
+        lifecycleExecutor = Executors.newSingleThreadScheduledExecutor(threadFactory);
+        state = State.STARTING;
+    }
+
+    /**
+     * Returns the lifecycle state of the {@link MetricsService} shared in application scope.
+     *
+     * @return lifecycle state of the shared {@link MetricsService}
+     */
+    public State getState() {
+        return state;
+    }
+
+    @PostConstruct
+    void init() {
+        lifecycleExecutor.submit(this::startMetricsService);
+        if (Boolean.parseBoolean(waitForService)
+            // "hawkular-metrics.backend" is not a real Metrics configuration parameter (there's a single
+            // MetricsService implementation, which is backed by Cassandra).
+            // But it's been used historically to wait for the service to be available before completing the deployment.
+            // Therefore, we still use it here for backward compatibililty.
+            // TODO remove when Hawkular build has been updated to use the eager startup flag
+            || "embedded_cass".equals(System.getProperty("hawkular-metrics.backend"))) {
+            long start = System.nanoTime();
+            while (state == State.STARTING
+                   // Give up after a minute. The deployment won't be failed and we'll continue to try to start the
+                   // service in the background.
+                   && NANOSECONDS.convert(1, MINUTES) > System.nanoTime() - start) {
+                Uninterruptibles.sleepUninterruptibly(1, SECONDS);
+            }
+        }
+    }
+
+    private void startMetricsService() {
+        if (state != State.STARTING) {
+            return;
+        }
+        LOG.info("Initializing metrics service");
+        connectionAttempts++;
+        try {
+            session = createSession();
+        } catch (Throwable t) {
+            Throwable rootCause = Throwables.getRootCause(t);
+            LOG.warn("Could not connect to Cassandra cluster - assuming its not up yet", rootCause);
+            // cycle between original and more wait time - avoid waiting huge amounts of time
+            long delay = 1 + ((connectionAttempts - 1) % 4);
+            LOG.warn("[{}] Retrying connecting to Cassandra cluster in [{}]s...", connectionAttempts, delay);
+            lifecycleExecutor.schedule(this::startMetricsService, delay, SECONDS);
+            return;
+        }
+        SchemaManager schemaManager = new SchemaManager(session);
```

Comment:
As I mentioned previously, I think the schema initialization should be done in ```MetricsServiceImpl```.  It is an implementation detail of the service. Right now the the CQL statements for the task service are in the same cql file, but  they will be separated out at some point as the task service schema is independent of the rest of the metrics schema.